supernode/p2p/service.go

```diff
@@ -0,0 +1,70 @@
+package p2p
+
+import (
+	"context"
+	"os"
+	"os/signal"
+
+	"github.com/pastelnetwork/gonode/common/log"
+)
+
+const (
+	logPrefix = "p2p"
+)
+
+// Service represents the p2p service.
+type Service struct {
+	dht DHT
+}
+
+// DHT represents the methods by which a library consumer interacts with a DHT.
+type DHT interface {
+	Store(data []byte, key []byte) (id string, err error)
+	Get(key string) (data []byte, found bool, err error)
+	CreateSocket() error
+	Listen() error
+	GetNetworkAddr() string
+	Bootstrap() error
+	UseStun() bool
+	Disconnect() error
+}
+
+// Run starts the DHT service
+func (service *Service) Run(ctx context.Context) error {
+	ctx = log.ContextWithPrefix(ctx, logPrefix)
+
+	err := service.dht.CreateSocket()
+	if err != nil {
+		panic(err)
+	}
+
+	go func() {
+		log.WithContext(ctx).Infof("Server listening on %q", service.dht.GetNetworkAddr())
+		err := service.dht.Listen()
+		panic(err)
+	}()
+
+	if err := service.dht.Bootstrap(); err != nil {
+		return err
+	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go func() {
+		for range c {
+			err := service.dht.Disconnect()
+			if err != nil {
+				panic(err)
+			}
+		}
+	}()
+
+	return nil
+}
+
+// NewService returns a new Service instance.
+func NewService(dht DHT) *Service {
```

Comment:
You don't need `supernode/p2p` at all. Keep everything in your folder `gonode/p2p` or move everything in `gonode/p2p`. It's up to you, but it should be in one place, for example `gonode/pastel`

Reply:
i separetd into supernode because it defines the interaction with supernode, contextully it belongs to supernode because it's a interface for interacting with p2p from supernode.

Reply:
I understand you, I was doing the same with `gonode/pastel` at the beginning, spit on two packages. 
But that doesn't make sense. only complicates everything. 
Please make decision to keep everything in `gonode/supernode/p2p` or `gonode/p2p`

Reply:
yeah you're right.